src/Firebase/Auth.php

```diff
@@ -479,4 +479,14 @@ public function unlinkProvider($uid, $provider): UserRecord
 
         return $this->getUser($uid);
     }
+
+    public function linkWithOauthCredentials($provider, string $token, bool $isAccessToken = true)
+    {
+        $provider = $provider instanceof Provider ? $provider : new Provider($provider);
+        $response = $this->client->signInWithOauthCredentials($provider, $token, $isAccessToken);
+
+        $data = JSON::decode((string) $response->getBody(), true);
+
+        return UserRecord::fromResponseData($data);
```

Comment:
Similar to the `unlinkProvider()` method I would prefer forwarding the execution to the `getUser()` method.

```php
$uid = JSON::decode((string) $response->getBody(), true)['localId'];

return $this->getUser($uid);
```

I know this will result in an additional request, but this way we can be sure that a returned User Record always results from the same logic (my focus in the SDK, in general, is more about correctness than about performance).

Reply:
I created a private function for this, because this two lines are used many times, so I think its a bit better this way.

Reply:
Good choice!